### PR TITLE
DOC: Add mention that mangle_dupe_cols is deprecated in user guide

### PR DESCRIPTION
**Pull request by phofl**
cc @datapythonista

### CODE REVIEW

File: doc/source/user_guide/io.rst [hunk 1]
```diff
@@ -611,6 +615,10 @@ If the header is in a row other than the first, pass the row number to
 Duplicate names parsing
 '''''''''''''''''''''''
 
+  .. deprecated:: 1.5.0
+     ``mangle_dupe_cols`` was never implemented, and a new argument where the
+     renaming pattern can be specified will be added instead.
+
```
Comment (datapythonista) [inline doc/source/user_guide/io.rst:621]: I'd personally update the section to just show that we rename columns, and remove all the references to `mangle_dupe_cols`, which never worked anyway.

**Review by datapythonista — COMMENTED** [review]
Thanks @phofl, good catch, I didn't know or expect to have about `mangle_dupe_cols` in the user guide, as this was never implemented. I propose one change, but looks good.

File: doc/source/user_guide/io.rst (hunk 1 above)
Comment (phofl) [reply doc/source/user_guide/io.rst:621]: Sounds good to me. Removed the example with False

**Review by mroeschke — APPROVED** [review]

**Comment by phofl** [comment]
@datapythonista ready to merge?

**Pull request merged**